### PR TITLE
Guided Transfer: Add host selection screen

### PR DESCRIPTION
**Pull request by jordwest**
This pull request adds the UI elements for the first step of Guided Transfer - selecting a host.

## How to test
1. Navigate to https://calypso.live/?branch=add/guided-transfer/host-select-screen
2. Click "My Sites" in the upper left corner of the page.
3. Scroll down in the left navigation area if needed and click "Settings"
4. Click "Export" in the page navigation.
5. Click "Purchase a Guided Transfer"

*Note: The new host selection buttons currently change the url hash to either `bluehost` or `siteground`. They will eventually lead to their respective pages.*

## What to expect

screen shot 2016-06-28 at 11 40 47 am

screen shot 2016-06-28 at 11 40 15 am

### CODE REVIEW

File: client/my-sites/guided-transfer/host-select.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+					<p>{ translate(
+`{{strong}}Please choose{{/strong}} one of our Guided Transfer compatible
+{{partner_link}}partner hosts{{/partner_link}}. You must have a hosting account
+with one of them to be able to move your site. Visit them {{lobby_link}}Guided
+Transfer Lobby{{/lobby_link}} if you have any question before starting, or
+{{learn_link}}learn more{{/learn_link}} about the process.`,
+						{
+							components: {
+								strong: <strong />,
+								partner_link: <a href="/" />,
```
Comment (dllh) [inline client/my-sites/guided-transfer/host-select.jsx]: https://get.wp.com/gt-hosting/

File: client/my-sites/guided-transfer/host-select.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+								lobby_link: <a href="/" />,
```
Comment (dllh) [inline client/my-sites/guided-transfer/host-select.jsx]: https://guidedtransfer.wordpress.com/

File: client/my-sites/guided-transfer/host-select.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+								learn_link: <a href="/" />,
```
Comment (dllh) [inline client/my-sites/guided-transfer/host-select.jsx]: https://en.support.wordpress.com/guided-transfer/

**Comment by dllh** [comment]
The images were missing for me in the calypso.live version. Maybe a quirk of the system. I ran the branch locally and it worked as expected. I've left comments with links that we can fill in where they're currently empty. Will leave code review to others, but functionally it works well.

File: client/my-sites/guided-transfer/host-select.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+							}
+						} ) }
+					</p>
+					<label>{ translate( 'Select your hosting provider' ) }</label>
```
Comment (jordwest) [inline client/my-sites/guided-transfer/host-select.jsx]: There was some discussion (p195om-2YV-p2) about "Select your hosting provider" being redundant, so we can remove this line.

File: client/my-sites/guided-transfer/style.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
+.guided-transfer {
```
Comment (jordwest) [inline client/my-sites/guided-transfer/style.scss]: As in #6200, the parent selector can be removed

File: client/my-sites/guided-transfer/host-select.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+					<div>
+						{ hostButtons.map( ( buttonOptions, index ) => {
+							return (
+								<Button className="guided-transfer__host-button" onClick={ buttonOptions.onClick } key={ index }>
```
Comment (jordwest) [inline client/my-sites/guided-transfer/host-select.jsx]: We should add an `aria-label` attribute to help screen readers

**Comment by jordwest** [comment]
This works for me. I'm wondering if there isn't somewhere else we could store the host logos - so that we're not checking them in and out of source control if hosts come and go.

I left a few other minor suggestions, but otherwise looks good.

**Comment by omarjackman** [comment]
@jordwest I handled all of the other comments but not sure where else to stick the partner logos other than on wp.com. @dllh any thoughts on that?

**Comment by dllh** [comment]
There's a Google apps partner image in `public/images/upgrades/`, so keeping these in a similar location makes sense to me. We could stick them in the `i` directory in the root of wordpress.com if you don't love checking them into this repo, but there's precedent for putting them in `public/images/upgrades/`, and I can't think of a reason not to commit them here.

File: client/my-sites/guided-transfer/host-select.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+import Button from 'components/button';
+
+class HostSelect extends Component {
+	showHost( hostSlug ) {
+		// Temporary page redirect with a # in it.
+		page( `/settings/export/${this.props.siteSlug}/guided#${hostSlug}` );
```
Comment (jordwest) [inline client/my-sites/guided-transfer/host-select.jsx]: We could probably change this to `/settings/export/guided/${hostSlug}/${this.props.siteSlug}` as per [this comment](https://github.com/Automattic/wp-calypso/pull/6200#discussion_r69292392)

**Comment by jordwest** [comment]
In that case, it's a 🚢  from me

**Comment by omarjackman** [comment]
@jordwest, I did a rebase which required some changes and I think this warrants a second review.

**Comment by jordwest** [comment]
Tested again, LGTM 👍

**Pull request merged**